realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -0,0 +1,511 @@
+            throw new IllegalStateException("A primary key is already defined");
+        }
+        table.setPrimaryKey(fieldName);
+        return this;
+    }
+
+    /**
+     * Removes the primary key from this class. This is the same as removing the {@link io.realm.annotations.PrimaryKey}
+     * annotation from the class.
+     *
+     * @return the updated schema.
+     * @throws IllegalArgumentException if the class doesn't have a primary key defined.
+     */
+    public RealmObjectSchema removePrimaryKey() {
+        if (!table.hasPrimaryKey()) {
+            throw new IllegalStateException(getClassName() + " doesn't have a primary key.");
+        }
+        table.setPrimaryKey("");
+        return this;
+    }
+
+    /**
+     * Sets a field to be required, i.e. not allowed to hold {@code null values}. This is equivalent to switching
+     * between boxed types and their primitive variant e.g. {@code Integer} to {@code int}.
+     *
+     * @param fieldName name of field in the class.
+     * @param required  {@code true} if field should be required, {@code false} otherwise.
+     * @return the updated schema.
+     * @throws IllegalArgumentException if field name doesn't exists or the field already has the given required flag.
+     * @see Required
+     */
+    public RealmObjectSchema setRequired(String fieldName, boolean required) {
+        long columnIndex = table.getColumnIndex(fieldName);
+        boolean currentColumnRequired = isRequired(fieldName);
+        RealmFieldType type = table.getColumnType(columnIndex);
+
+        if (type == RealmFieldType.OBJECT) {
+            throw new IllegalArgumentException("Cannot modify the required state for RealmObject references: " + fieldName);
+        }
+        if (type == RealmFieldType.LIST) {
+            throw new IllegalArgumentException("Cannot modify the required state for RealmList references: " + fieldName);
+        }
+        if (required && currentColumnRequired) {
+            throw new IllegalStateException("Field is already required: " + fieldName);
+        }
+        if (!required && !currentColumnRequired) {
+            throw new IllegalStateException("Field is already nullable: " + fieldName);
+        }
+
+        if (required) {
+            table.convertColumnToNotNullable(columnIndex);
+        } else {
+            table.convertColumnToNullable(columnIndex);
+        }
+        return this;
+    }
+
+    /**
+     * Sets a field to be nullable, i.e. it should be able to hold {@code null values}. This is equivalent to switching
+     * between primitive types and their boxed variant e.g. {@code int} to {@code Integer}.
+     *
+     * @param fieldName name of field in the class.
+     * @param nullable  {@code true} if field should be nullable, {@code false} otherwise.
+     * @return the updated schema.
+     * @throws IllegalArgumentException if field name doesn't exists or the field already has the given nullable flag.
```

Comment:
if field name doesn't exists, cannot be applied with @{@link Required} or the field already has the given nullable flag.